### DISCUSSION TITLE
[@types/react] `JSX.IntrinsicElements` from `@types/react` Does Not Support MathML Elements

### DISCUSSION THREAD

**Discussion by louisscruz** [DefinitelyTyped/DefinitelyTyped#69884] [Issues with a @types package]
[MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) specifies a set of elements that can be used to describe math expressions in an XML format. This is now supported in modern versions of [Chrome, Safari, Opera, Firefox, and Edge](https://caniuse.com/mathml). When using the current latest versions of `@types/react`, if one tries to use any MathML elements, one sees something like:

<img width="572" alt="Screenshot 2024-06-22 at 8 08 22 PM" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/7310102/3a8981e0-c6dc-43e9-b5e2-904bb2e92a90">

Is it possible to add support for MathML elements? I would be interested in contributing to make this possible, but I would like to double check if that is acceptable first.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
If it works at runtime, we should support it with types. Have you checked if all of these elements and their props work at runtime?

**Comment by louisscruz** [comment]
Yes, it works. The `math` example I gave was lifted from https://developer.mozilla.org/en-US/docs/Web/MathML/Element/math, where you can see it working in the browser:

<img width="1170" alt="Screenshot 2024-06-24 at 5 59 31 PM" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/7310102/6a727c97-1ba6-456a-9b45-8fdd6e0cbb51">

I also tested that `math` working in a brand new Next project. Also, I'm testing in Chrome 126.

It looks like some of the elements and attributes of elements related to MathML have varying degrees of support across browsers, and quite a few are marked as legacy/deprecated. Should legacy/deprecated elements/attributes be excluded? The support for them seems much spottier.

If I open a PR to add this, would that be considered? It's a bit tedious, but I can make a pass to add the relevant `math` elements + attributes from the [various MathML elements](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/math) listed on the MDN page.

**Reply by eps1lon** [reply]
> If I open a PR to add this, would that be considered?

Sure.

**Comment by louisscruz** [comment]
I've opened a PR to add initial MathML support here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69986
